src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -229,8 +226,15 @@ ConcurrencyWorker::ResetFreeCtxIds()
 {
   std::lock_guard<std::mutex> lock(cb_mtx_);
   free_ctx_ids_ = std::queue<int>();
-  for (size_t i = 0; i < ctxs_.size(); ++i) {
-    free_ctx_ids_.push(i);
+
+  if (ctxs_.size() > 1) {
```

Comment:
Why when there's 2 or more contexts?

Reply:
I had actually already cleaned this up some more in the follow-on story. I've pulled that change back into this story for clarity.